src/withStyles.jsx

```diff
@@ -68,12 +71,46 @@ export function withStyles(
     // NOTE: Use a class here so components are ref-able if need be:
     // eslint-disable-next-line react/prefer-stateless-function
     class WithStyles extends BaseClass {
+      constructor(props, context) {
+        super(props, context);
+
+        // direction needs to be stored in state in order to trigger a rerender
+        // when context changes.
+        this.state = {
+          direction: context[CHANNEL] ? context[CHANNEL].getState() : defaultDirection,
+        };
+      }
+
       componentWillMount() {
         this.maybeCreateStyles();
       }
 
+      componentDidMount() {
+        if (this.context[CHANNEL]) {
+          // subscribe to future direction changes
+          this.channelUnsubscribe = this.context[CHANNEL].subscribe((direction) => {
```

Comment:
could you remind me, did we consider using the `withDirection` HOC directly instead of reimplementing this?

Reply:
I considered it, but adding the extra virtual DOM node seemed possibly problematic. We can go back and do that in a patch change if that makes more sense moving forward.

I did think it was sad to duplicate this code.... so maybe the right thing to do would be to wrap?

I'll defer to @lencioni when he is back from PTO.

Reply:
Because this HOC is used so prolifically throughout a tree, I would be hesitant to drop in another component into the mix. If you wanted to keep it DRY you could probably export some utility methods from withDirection and then use them here.

Reply:
That’d be a great follow up, i had the same thought